internal/engine/worker/contract/ens/worker.go

```diff
@@ -0,0 +1,556 @@
+package ens
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/crypto/secp256k1"
+	"github.com/rss3-network/serving-node/config"
+	"github.com/rss3-network/serving-node/internal/database"
+	"github.com/rss3-network/serving-node/internal/database/model"
+	"github.com/rss3-network/serving-node/internal/engine"
+	source "github.com/rss3-network/serving-node/internal/engine/source/ethereum"
+	"github.com/rss3-network/serving-node/provider/ethereum"
+	"github.com/rss3-network/serving-node/provider/ethereum/contract/ens"
+	"github.com/rss3-network/serving-node/provider/ethereum/contract/erc1155"
+	"github.com/rss3-network/serving-node/provider/ethereum/contract/erc20"
+	"github.com/rss3-network/serving-node/provider/ethereum/contract/erc721"
+	"github.com/rss3-network/serving-node/provider/ethereum/token"
+	"github.com/rss3-network/serving-node/schema"
+	"github.com/rss3-network/serving-node/schema/filter"
+	"github.com/rss3-network/serving-node/schema/metadata"
+	"github.com/samber/lo"
+	"github.com/shopspring/decimal"
+)
+
+// Worker is the worker for ENS.
+var _ engine.Worker = (*worker)(nil)
+
+type worker struct {
+	// Base
+	config         *config.Module
+	ethereumClient ethereum.Client
+	tokenClient    token.Client
+	databaseClient database.Client
+	// Specific filters
+	baseRegistrarImplementationFilterer *ens.BaseRegistrarImplementationFilterer
+	ethRegistrarControllerV1Filterer    *ens.ETHRegistrarControllerV1Filterer
```

Comment:
Ambiguous naming, ethRegistrarControllerV1Filterer or ensRegistrarControllerV1Filterer?

Reply:
It's called `ENS: ETH Registrar Controller` in etherscan tag (or `Old` for V1), so it should be fine.